### ISSUE TITLE
Bug in CTRL generation

### ISSUE THREAD

**Issue by cfoster0** [huggingface/transformers#1473]
## 🐛 Bug

Model: CTRL

Language: English

The problem arises when using:
* [x] Official example script [`run_generation.py`](https://github.com/huggingface/transformers/blob/master/examples/run_generation.py)

The tasks I am working on is:
* [x] Generating text with the CTRL model.

## To Reproduce

1. Run `python run_generation.py` with `model_type=ctrl`, `model_name_or_path=ctrl`, `temperature=0`, a decent length like `length=50`, and `repetition_penalty=1.2`.
2. Input a `Links`-based control code, such as `Links https://www.cnn.com/2018/09/20/us-president-meets-british-pm`, from the original paper.

Rather than generating relevant, English text, it will often generate assorted, garbled French. For example, for the above example it generated: `m m e et au neuen auge de la part des 
 \* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *`.

## Expected behavior

It should generate relatively-coherent English text, relevant to the link URL. This is the behavior in the paper, and in the [`lower_memory` branch](https://github.com/salesforce/ctrl/tree/lower_memory) Colab notebook.

## Environment

* OS: MacOS
* Python version: 3.5.6, Anaconda.
* PyTorch version: 1.2.0
* PyTorch Transformers version (or branch): 2.1.0
* Using GPU ? No
* Any other relevant information:

**Comment by LysandreJik** [comment]
Yes we have observed there was a difference in tokenization. We've temporarily fixed in 036483f, could you install from source and tell us if you manage to have good generations?

By following the recommended specs (temperature=0.2, top_k=5 and repetition_penalty=1.2), with the following input sentence: `Reviews Rating 4.0`, I obtained the following completion:

Reviews Rating 4.0 < GENERATION > "out of 5 stars. I received a copy from the author in exchange for an honest review. 
Rating: 4.0 
 This is one book that you will not want to put down. It was very well written and kept me on my toes. The characters were so real it made them seem like people you know. You could feel their pain as they struggled with what happened to them. There are some twists and turns along the way but nothing too surprising. All in all this story had everything needed to make it a great book. If"

**Comment by cfoster0** [comment]
Looks like that fixed it. Getting a similar completion to what you got for `Reviews Rating 4.0`:

```
Reviews Rating 4.0 out of 5 starsI received a copy from the author in exchange for an honest review. 
 Rating: 4.0 
 This was a great book that kept me interested and wanting to read more. It is about two people who have been together forever but are not
```

And for `Links https://www.cnn.com/2018/09/20/us-president-meets-british-pm`, I get:

```
Links https://www.cnn.com/2018/09/20/us-president-meets-british-pm (CNN)President Donald Trump said Friday he would meet with British Prime Minister Theresa May in Washington next week, a move that could help ease tensions between the two countries after months of escalating trade tensions. 
 
 The White House announced Trump's decision to hold talks
```

Should I leave this issue open, since you mentioned this is a temporary fix?
